packages/block-library/src/image/index.php

```diff
@@ -32,45 +32,47 @@ function render_block_core_image( $attributes, $content, $block ) {
 		$processor->set_attribute( 'data-id', $attributes['data-id'] );
 	}
 
-	$lightbox_enabled  = false;
 	$link_destination  = isset( $attributes['linkDestination'] ) ? $attributes['linkDestination'] : 'none';
 	$lightbox_settings = block_core_image_get_lightbox_settings( $block->parsed_block );
 
-	// If the lightbox is enabled and the image is not linked, flag the lightbox to be rendered.
-	if ( isset( $lightbox_settings ) && 'none' === $link_destination ) {
+	$view_js_file_handle = 'wp-block-image-view';
+	$script_handles      = $block->block_type->view_script_handles;
 
-		if ( isset( $lightbox_settings['enabled'] ) && true === $lightbox_settings['enabled'] ) {
-			$lightbox_enabled = true;
-		}
-	}
-
-	// If at least one block in the page has the lightbox, mark the block type as interactive.
-	if ( $lightbox_enabled ) {
+	/*
+	 * If the lightbox is enabled and the image is not linked, add the filter
+	 * and the JavaScript view file.
+	 */
+	if (
+		isset( $lightbox_settings ) &&
+		'none' === $link_destination &&
+		isset( $lightbox_settings['enabled'] ) &&
+		true === $lightbox_settings['enabled']
+	) {
 		$block->block_type->supports['interactivity'] = true;
```

Comment:
@artemiomorales @SantosGuillamot when you have a chance, I'd appreciate your feedback on this line of code:
- I have no idea what it does.
- Is it really necessary?
- If it is, should it be changed back to `false` in the `else` statement?

Reply:
From what I understand, that line moves the interactivity scripts to the footer, which is a helpful optimization.
It shouldn't be changed to `false`, because it should be `true` so long as any image has the lightbox enabled.

Pinging @luisherranz to see if he'd like to detail better! In any case, I don't believe a change here is a blocker for this PR.

Reply:
We're actually thinking about removing the conditionality and use `supports.interactivity: true` for any block that _might_ be interactive at some point. 

But we also need to figure out how those conditional interactive blocks can declare that conditionality, or even be able to combine different interactive behaviors depending on their configuration.

Related conversation here: 
- https://github.com/WordPress/gutenberg/discussions/54194